src/rust-crypto/rust-crypto.ts

```diff
@@ -1641,6 +1654,63 @@ export class RustCrypto extends TypedEventEmitter<RustCryptoEvents, RustCryptoEv
     public async getOwnIdentity(): Promise<RustSdkCryptoJs.OwnUserIdentity | undefined> {
         return await this.olmMachine.getIdentity(new RustSdkCryptoJs.UserId(this.userId));
     }
+
+    /**
+     * Returns whether dehydrated devices are supported by the crypto backend
+     * and by the server.
+     */
+    public async isDehydrationSupported(): Promise<boolean> {
+        return await this.dehydrationManager.isSupported();
+    }
+
+    /**
+     * Rehydrate the dehydrated device stored on the server
+     *
+     * Checks if there is a dehydrated device on the server.  If so, rehydrates
+     * the device and processes the to-device events.
+     *
+     * Returns whether or not a dehydrated device was found.
+     */
+    public async rehydrateDeviceIfAvailable(): Promise<boolean> {
+        return await this.dehydrationManager.rehydrateDeviceIfAvailable();
+    }
+
+    /**
+     * Creates and uploads a new dehydrated device
+     *
+     * Creates and stores a new key in secret storage if none is available.
+     */
+    public async createAndUploadDehydratedDevice(): Promise<void> {
+        return await this.dehydrationManager.createAndUploadDehydratedDevice();
+    }
+
+    /** Return whether the dehydration key is stored in Secret Storage
+     */
+    public async isDehydrationKeyStored(): Promise<boolean> {
+        return await this.dehydrationManager.isKeyStored();
+    }
+
+    /**
+     * Schedule periodic creation of dehydrated devices
+     *
+     * @param interval - the time to wait between creating dehydrated devices
+     * @param delay - how long to wait before creating the first dehydrated device.
+     *     Defaults to creating the device immediately.
+     */
+    public async scheduleDeviceDehydration(interval: number, delay?: number): Promise<void> {
+        return await this.dehydrationManager.scheduleDeviceDehydration(interval, delay);
+    }
+
+    /**
+     * Reset the dehydration key
+     *
+     * Note: this does not create a new dehydrated device.  This will need to
+     * be done either by calling `createAndUploadDehydratedDevice` or
+     * `scheduleDeviceDehydration`.
+     */
+    public async resetDehydrationKey(): Promise<void> {
+        return await this.dehydrationManager.resetKey();
+    }
```

Comment:
Could you move all these to the `CryptoApi implementation` section of the class?